gems/pending/spec/util/mount/miq_generic_mount_session_spec.rb

```diff
@@ -2,25 +2,29 @@
 require 'util/mount/miq_generic_mount_session'
 
 describe MiqGenericMountSession do
-  context "#connect" do
-    before do
-      MiqGenericMountSession.stub(:raw_disconnect)
-      @s1 = MiqGenericMountSession.new({:uri => '/tmp/abc'})
-      @s2 = MiqGenericMountSession.new({:uri => '/tmp/abc'})
+  let(:prefix) { File.join(Dir.tmpdir, "miq_") }
 
-      @s1.logger = Logger.new("/dev/null")
-      @s2.logger = Logger.new("/dev/null")
+  it "#connect returns a string pointing to the mount point" do
+    described_class.stub(:raw_disconnect)
+    s = described_class.new(:uri => '/tmp/abc')
+    s.logger = Logger.new("/dev/null")
+
+    expect(s.connect).to match(%r(\A#{prefix}\d{8}-\d{5}-\w+\z))
+
+    s.disconnect
+  end
+
+  context "#mount_share" do
+    it "without :mount_point uses default temp directory as a base" do
+      expect(described_class.new(:uri => '/tmp/abc').mount_share).to match(%r(\A#{prefix}\d{8}-\d{5}-\w+\z))
```

Comment:
We're concerned with everything up to the `_` since you have the ability to set a directory or not.